src/framework/pybind_json.hpp

```diff
@@ -0,0 +1,268 @@
+ * copyright notice, and modified files need to carry a notice indicating
+ * that they have been altered from the originals.
+ */
+
+#ifndef _aer_framework_pybind_json_hpp_
+#define _aer_framework_pybind_json_hpp_
+
+#include <complex>
+#include <cstdint>
+#include <fstream>
+#include <iostream>
+#include <map>
+#include <vector>
+
+#include <pybind11/pybind11.h>
+#include <pybind11/cast.h>
+#include <pybind11/stl.h>
+#include <pybind11/numpy.h>
+#include <pybind11/complex.h>
+#include <iostream>
+#include <type_traits>
+
+#include <nlohmann_json.hpp>
+#include "framework/matrix.hpp"
+
+namespace py = pybind11;
+namespace nl = nlohmann;
+using namespace pybind11::literals;
+using json_t = nlohmann::json;
+
+//============================================================================
+// JSON Conversion for pybind types
+//============================================================================
+
+namespace std {
+
+/**
+ * Convert a python object to a json.
+ * @param js a json_t object to contain converted type.
+ * @param o is a python object to convert.
+ */
+void to_json(json_t &js, const py::handle &o);
+
+/**
+ * Create a python object from a json
+ * @param js a json_t object 
+ * @param o is a reference to an existing (empty) python object
+ */
+void from_json(const json_t &js, py::object &o);
+
+} // end namespace std.
+
+/**
+ * Convert a numpy array to a json object
+ * @param arr is a numpy array 
+ * @returns a json list (potentially of lists)
+ */
+template <typename T>
+json_t numpy_to_json(py::array_t<T, py::array::c_style> arr);
+
+/**
+ * Convert a 1-d numpy array to a json object
+ * @param arr is a numpy array 
+ * @returns a json list (potentially of lists)
+ */
+template <typename T>
+json_t numpy_to_json_1d(py::array_t<T, py::array::c_style> arr);
+
+/**
+ * Convert a 2-d numpy array to a json object
+ * @param arr is a numpy array 
+ * @returns a json list (potentially of lists)
+ */
+template <typename T>
+json_t numpy_to_json_2d(py::array_t<T, py::array::c_style> arr);
+
+/**
+ * Convert a 3-d numpy array to a json object
+ * @param arr is a numpy array 
+ * @returns a json list (potentially of lists)
+ */
+template <typename T>
+json_t numpy_to_json_3d(py::array_t<T, py::array::c_style> arr);
+
+/*******************************************************************************
+ *
+ * Implementations
+ *
+ ******************************************************************************/
+
+//------------------------------------------------------------------------------
+// JSON Conversion
+//------------------------------------------------------------------------------
+
+template <typename T>
+json_t numpy_to_json_1d(py::array_t<T, py::array::c_style> arr) {
+    py::buffer_info buf = arr.request();
+    if (buf.ndim != 1) {
+        throw std::runtime_error("Number of dims must be 1");
+    }
+
+    T *ptr = (T *) buf.ptr;
+    size_t D0 = buf.shape[0];
+
+    std::vector<T> tbr; // to be returned 
```

Comment:
oh, ok. There's no need for a comment if you call the variable: `to_be_returned` instead of ``tbr` :) see: https://en.wikipedia.org/wiki/Self-documenting_code
I think naminig it `ret` is fine, it's like a standard-de-facto.